### ISSUE TITLE
[REQUEST] Filter and Read Progress in Mangadex History

### ISSUE THREAD

**Issue by Zw012d** [Glagan/SyncDex#39]
I have been using `Follows `-> `Latest updates` for checking latest chapters of mangas that i have read but due to limitation of Mangadex not having infinite scrolling we get multiple blank pages (image below) while using Syncdex features and it is a pain to check all the pages when having a large list.

![Screenshot 2021-01-30 110644](https://user-images.githubusercontent.com/76645315/106348323-784bc400-62eb-11eb-9b15-ead2a3343e36.png)

One solution apart from changes in Mangadex UI itself would be having filtering (2 in image below) in history page based on `Higher chapter available` / `Latest chapter read` / `All` as it has infinite scrolling. We can use sort if filtering is not feasible to separate `Higher chapter available` and `Latest chapter read`. Currently we can visually distinguish `Higher chapter available` in history (Blue shade below manga tile) for updates but in case of large history it is not viable. Also having Read Progress (for illustration purpose 3 or 4 in image below) based on last read chapter and total chapters available would be very helpful and useful feature.

![Screenshot 2021-01-30 101718](https://user-images.githubusercontent.com/76645315/106349452-eb593880-62f3-11eb-9be4-7d7c400d8163.png)

On the side note `Your last xx opened titles are listed below.`  (1 in image above) doesn't reflect the manga list in history currently, would it be fixed by itself after repopulating history as history disappear bug is fixed now.

**Comment by Glagan** [comment]
if I can summarize you request it's to have a filter in the history page to only show one or the other, and a progress indicator for the latest available chapter on MangaDex ?  
For the list with infinite scrolling, I'm not sure if it should be on my end to make it, with what I can do I don't know if it would really make an improvement, as I would just load the next page, append it and highlight it. 

I will surely add the filter and the progress indicator in the future, but not in the next release, I'm currently in the middle of a refactor that touch a lot of components and I'm trying to avoid to change things that's related to it, and I already did with some previous fixes.

**Comment by Zw012d** [comment]
> if I can summarize you request it's to have a filter in the history page to only show one or the other, and a progress indicator for the latest available chapter on MangaDex ?

Yes, filter in the history page based on `Higher chapter available` / `Latest chapter read` / `All` (All shows all history) and a simple progress indicator ex. - last read chapter (eng/same language) of a manga is 10 and highest chapter available (eng/same language) is 20 then some progress indicator showing 50%.

> For the list with infinite scrolling, I'm not sure if it should be on my end to make it, with what I can do I don't know if it would really make an improvement, as I would just load the next page, append it and highlight it.

Exactly that's why i requested additional features in history page as we already doing improvements in history page.

> I will surely add the filter and the progress indicator in the future, but not in the next release, I'm currently in the middle of a refactor that touch a lot of components and I'm trying to avoid to change things that's related to it, and I already did with some previous fixes.

Thanks i will wait patiently for the future release :)
